test/functional/p2p_quorum_data.py

```diff
@@ -0,0 +1,410 @@
+                self.sync_blocks()
+
+        self.nodes[0].spork("SPORK_17_QUORUM_DKG_ENABLED", 0)
+        self.nodes[0].spork("SPORK_19_CHAINLOCKS_ENABLED", 4070908800)
+
+        self.wait_for_sporks_same()
+        quorum_hash = self.mine_quorum()
+
+        node0 = self.nodes[0]
+        mn1 = self.mninfo[0]
+        mn2 = self.mninfo[1]
+        mn3 = self.mninfo[2]
+
+        fake_mnauth_1 = ["cecf37bf0ec05d2d22cb8227f88074bb882b94cd2081ba318a5a444b1b15b9fd",
+                    "087ba00bf61135f3860c4944a0debabe186ef82628fbe4ceaed1ad51d672c58dde14ea4b321efe0b89257a40322bc972"]
+        fake_mnauth_2 = ["6ad7ed7a2d6c2c1db30fc364114602b36b2730a9aa96d8f11f1871a9cee37378",
+                    "122463411a86362966a5161805f24cf6a0eef08a586b8e00c4f0ad0b084c5bb3f5c9a60ee5ffc78db2313897e3ab2223"]
+
+        p2p_node0 = p2p_connection(node0)
+        p2p_mn1 = p2p_connection(mn1.node)
+        network_thread_start()
+        p2p_node0.wait_for_verack()
+        p2p_mn1.wait_for_verack()
+        id_p2p_node0 = get_mininode_id(node0)
+        id_p2p_mn1 = get_mininode_id(mn1.node)
+
+        wait_for_banscore(node0, id_p2p_node0, 0)
+        wait_for_banscore(mn1.node, id_p2p_mn1, 0)
+
+        quorum_hash_int = int(quorum_hash, 16)
+        protx_hash_int = int(mn1.proTxHash, 16)
+
+        # Valid requests
+        qgetdata_vvec = msg_qgetdata(quorum_hash_int, 100, 0x01, protx_hash_int)
+        qgetdata_contributions = msg_qgetdata(quorum_hash_int, 100, 0x02, protx_hash_int)
+        qgetdata_all = msg_qgetdata(quorum_hash_int, 100, 0x03, protx_hash_int)
+        # The normal node should not respond to qgetdata
+        p2p_node0.test_qgetdata(qgetdata_all, response_expected=False)
+        wait_for_banscore(node0, id_p2p_node0, 10)
+        # The masternode should not respond to qgetdata for non-masternode connections
+        p2p_mn1.test_qgetdata(qgetdata_all, response_expected=False)
+        wait_for_banscore(mn1.node, id_p2p_mn1, 10)
+        # Open a fake MNAUTH authenticated P2P connection to the masternode to allow qgetdata
+        node0.disconnect_p2ps()
+        mn1.node.disconnect_p2ps()
+        network_thread_join()
+        p2p_mn1 = p2p_connection(mn1.node)
+        network_thread_start()
+        p2p_mn1.wait_for_verack()
+        id_p2p_mn1 = get_mininode_id(mn1.node)
+        mnauth(mn1.node, id_p2p_mn1, fake_mnauth_1[0], fake_mnauth_1[1])
+        # The masternode should now respond to qgetdata requests
+        # Request verification vector
+        p2p_mn1.test_qgetdata(qgetdata_vvec, 0, self.llmq_threshold, 0)
+        wait_for_banscore(mn1.node, id_p2p_mn1, 0)
+        # Request encrypted contributions
+        p2p_mn1.test_qgetdata(qgetdata_contributions, 0, 0, self.llmq_size)
+        wait_for_banscore(mn1.node, id_p2p_mn1, 25)
+        # Request both
+        p2p_mn1.test_qgetdata(qgetdata_all, 0, self.llmq_threshold, self.llmq_size)
+        wait_for_banscore(mn1.node, id_p2p_mn1, 50)
```

Comment:
Why does this bump ban score?

Reply:
I requests more than once within `EXPIRATION_TIMEOUT` window https://github.com/dashpay/dash/blob/0b0a7a0ec6f8e4f9bb1167c9f1ff4b923163d03d/src/llmq/quorums.cpp#L500